d3ledger/back-office
# PDF/CSV report functionality

# Ticket
D3-118

# Description
Adds report functionality.

The main part is report-util.js, which generates PDF/CSV based on parameters. `generatePDF` and `generateCSV` calls `generateReportData` internally, which generates JSON data to be put on a report.

* report-util.js
  * [x] calculate figures to put on a report (generateReportData)
  * [x] PDF output (generatePDF)
  * [x] CSV output (generateCSV)
* ReportPage.vue
  * [x] make the report page available
* Tests

# How to check
1. `yarn` to install additional dependencies.
2. go to the report page and check if the previous month's reports are listed.
3. click "New Report" and try some.

## example outputs
[report-ethereum-20180701-20180717.pdf](https://github.com/d3ledger/back-office/files/2222641/report-ethereum-20180701-20180717.pdf)
[report-eos-20180701-20180731.csv.txt](https://github.com/d3ledger/back-office/files/2248156/report-eos-20180701-20180731.csv.txt)
